### PR TITLE
Première version du jeu de dame

### PR DESCRIPTION
**Pull request by soursier**
Compile via cmake

### CODE REVIEW

File: lib/dames.c [hunk 1]
```diff
@@ -17,7 +28,13 @@ void init_table(pion grille[][WIDTH_TABLE]) {
 }
 
 void print_board(pion grille[][WIDTH_TABLE]) {
```
Comment (agautier2) [inline lib/dames.c:30]: what's mean "grille", do you want to say grill ? to cook the meat

File: README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+un fichier make file contenant 
+```
+CC ?= gcc
+CFLAGS += -Wall --pedantic -Werror
+
+all:
+	$(CC) $(LDFLAGS) -o dames $(CFLAGS) lib/dames.c main.c
+```
+devrait fonctionner
```
Comment (ycardaillac) [inline README.md:11]: Probably...?

File: lib/dames.c [hunk 3]
```diff
@@ -2,6 +2,17 @@
...
 
+void game() {
+  pion board[WIDTH_TABLE][HEIGHT_TABLE];
+
+  init_table(board);
+  while (true) {
+    print_board(board);
+    player_turn(board, PLAYER_WHITE);
+    print_board(board);
+    player_turn(board, PLAYER_BLACK);
```
Comment (ycardaillac) [inline lib/dames.c:13]: nice way to avoid having to change the player variable!

File: lib/dames.c [hunk 4]
```diff
@@ -35,59 +52,62 @@ void print_board(pion grille[][WIDTH_TABLE]) {
...
-      printf("Ceci n'est pas votre pion !\n");
-    } else if (grille[y][x] == PION_EMPTY) {
-      printf("Cet endroit est vide ! \n");
-    } else {
-      isValid = true;
-      printf("valeur du pion %d", grille[y][x]);
+    for (int moves_index = 0; moves_index < nb_moves; moves_index++) {
+      *dest = possible_moves[moves_index].destination;
+      temp_piece = possible_moves[moves_index].piece;
+      printf("la piece x = %d y = %d peut elle aller en x=%d y=%d ?\n", piece.x,
```
Comment (ycardaillac) [inline lib/dames.c:100]: that's debug right?

File: lib/dames.c [hunk 5]
```diff
@@ -35,59 +52,62 @@ void print_board(pion grille[][WIDTH_TABLE]) {
...
+             piece.y, x, y);
+      printf("celle ci x = %d y = %d peut aller en x = %d y = %d\n",
```
Comment (ycardaillac) [inline lib/dames.c:102]: debug?

Comment (ycardaillac) [reply lib/dames.c:102]: if you want to leave debug log, use a logger or a macro that can disable logs.

File: lib/dames.c [hunk 6]
```diff
@@ -141,27 +196,65 @@ void check_moves_piece(pion grille[][WIDTH_TABLE], player current_player,
...
 
-void check_eat(pion grille[][WIDTH_TABLE], player current_player, pos pion,
-               move possible_moves[], int* nbMoves, bool* can_eat) {
-  pos truc = machin;
+void check_is_eating_move(pion grille[][WIDTH_TABLE], player current_player,
+                          pos pion, pos possible_ennemy, move possible_moves[],
+                          int* nbMoves, bool* can_eat) {
+  pos dest;
+  dest.y = 2 * possible_ennemy.y - pion.y;
+  dest.x = 2 * possible_ennemy.x - pion.x;
```
Comment (ycardaillac) [inline lib/dames.c:204]: ahaha nice one, everybody has it's own way of finding this "dx" "dy" element.

File: main.c [hunk 7]
```diff
@@ -4,16 +4,6 @@
 #include "lib/dames.h"
```
Comment (ycardaillac) [inline main.c:4]: Generally speaking, you would be having, a src dir with all .c, a include dir with .h.

And if needed a directory lib with external libs has .so or sources (even though that's bad practice as project are generally better kept separated.)

**Pull request merged**